lib/cc/services/asana.rb

```diff
@@ -14,21 +14,24 @@ class Config < CC::Service::Config
     validates :workspace_id, presence: true
   end
 
+  ENDPOINT = "https://app.asana.com/api/1.0/tasks"
+
   self.title = "Asana"
   self.description = "Create tasks in Asana"
   self.issue_tracker = true
 
   def receive_test
     result = create_task("Test task from Code Climate")
-
-    {
-      ok: true,
-      message: "Ticked <a href='#{result[:url]}'>#{result[:id]}</a> created."
-    }
-  rescue => ex
-    { ok: false, message: ex.message }
+    result.merge(
+      message: "Ticket <a href='#{result[:url]}'>#{result[:id]}</a> created."
+    )
+  rescue CC::Service::HTTPError => ex
+    body = JSON.parse(ex.response_body)
+    ex.user_message = body["errors"].map{|e| e["message"] }.join(" ")
+    raise ex
```

Comment:
It's no so much logic as it is that we need to augment the exception. There's information that the `receive_test` methods need to put into the replies that can't be handled any other way. That's why we're re-raising the error -- the `WithErrorHandling` middleware handles the errors just fine, but the since the `receive_test` handlers are user-facing, they want/need to put pretty messages on the data returned.

It's possible we also want to handle this in the web-side, though that will be a bit more complicated.